### DISCUSSION TITLE
Password protected _site

### DISCUSSION THREAD

**Discussion by christoph-LSN** [open-sdg/open-sdg#1507] [Show and tell]
@brockfanning @LucyGwilliamAdmin @MoBosse 

Hello, 

my clients asked me to have a password protection for the site during the testing time when it's published on the custom domain but not yet approved from all people that are involved with the whole thing.

Searching with Google I found this: https://www.tachyonstemplates.com/2020/jekyll-netlify-password/

I modified this line:

https://github.com/christoph-LSN/MT_Site/blob/11d460267d52f732c9cacccea185a023986a04b7/.github/workflows/deploy-to-staging.yml#L32

and as you can see here https://christoph-lsn.github.io/MT_Site/ the site is protected.

The password is: IM_TEST
fill in and the site will be decrypted.

Two questions: 

1) Yes, this is some kind of a hard hack... I would like to have it more elegant and implement it in the site_config like:

pwd_protection: yes/no
pwd: IM_TEST

But I don't now how to use variables in the deploy-to-stanging file..

2) Do you think this will be worth to inplement in open-sdg?

I have tested it in chrome, firefox and Micrsoft edge

**Comment by MoBosse** [comment]
Pretty cool. I used to experiment with a similar approach: [https://github.com/matteobrusa/Password-protection-for-static-pages](https://github.com/matteobrusa/Password-protection-for-static-pages). 

I can say that we would actually use such a feature, only we would need to have the ability to set a password for each indicator page. The background is that sometimes we want data providers to see the result before we publish their data. The problem is that we don't want to communicate one pre-published page to all the different data providers so that they can't see the other providers' data that hasn't been published yet. Password protecting the individual indicator pages would be a good solution to this.

**Reply by christoph-LSN** [reply]
I have had also took  a look at this [https://github.com/lilykonings/jekyll-password-protect](https://github.com/lilykonings/jekyll-password-protect),  but I don't know what's the  `SCR-FOLDER` and the `DEST-FOLDER` and where to put the `gulpfile.js`
